hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -0,0 +1,533 @@
+
+    async def _request_order_update(self, order: InFlightOrder) -> Dict[str, Any]:
+        raise Exception("request order update not yet implemented")
+
+    async def _request_order_fills(self, order: InFlightOrder) -> Dict[str, Any]:
+        raise Exception("request order fills not yet implemented")
+
+    async def _all_trade_updates_for_order(self, order: InFlightOrder) -> List[TradeUpdate]:
+        blxr_client_order_i_d = convert_hummingbot_to_blxr_client_order_id(order.client_order_id)
+        await asyncio.sleep(2)
+        order_updates = self._order_manager.get_order_status(
+            trading_pair=order.trading_pair, client_order_id=blxr_client_order_i_d
+        )
+        trade_updates = []
+        for order_update in order_updates:
+            if (
+                order_update.order_status == OrderStatus.OS_FILLED
+                or order_update.order_status == OrderStatus.OS_PARTIAL_FILL
+            ):
+                side = order_update.side
+                fill_price = Decimal(order_update.fill_price)
+                fill_base_amount: Decimal = Decimal(0)
+                fill_quote_amount: Decimal = Decimal(0)
+
+                if side == Side.S_ASK:
+                    fill_base_amount = Decimal(order_update.quantity_released)
+                    fill_quote_amount = Decimal(fill_base_amount) * fill_price
+                elif side == Side.S_BID:
+                    fill_quote_amount = Decimal(order_update.quantity_released)
+                    fill_base_amount = Decimal(fill_quote_amount) * (1 / fill_price)
+
+                fee = TradeFeeBase.new_spot_fee(
+                    fee_schema=self.trade_fee_schema(),
+                    trade_type=order.trade_type,
+                )
+                trade_updates.append(
+                    TradeUpdate(
+                        trade_id=order.client_order_id,
+                        client_order_id=order.client_order_id,
+                        exchange_order_id=order.exchange_order_id,
+                        trading_pair=order.trading_pair,
+                        fill_timestamp=order_update.timestamp,
+                        fill_price=fill_price,
+                        fill_base_amount=fill_base_amount,
+                        fill_quote_amount=fill_quote_amount,
+                        fee=fee,
+                    )
+                )
+
+        return trade_updates
+
+    async def _request_order_status(self, tracked_order: InFlightOrder) -> OrderUpdate:
+        blxr_client_order_id = convert_hummingbot_to_blxr_client_order_id(tracked_order.client_order_id)
+        order_status_info = self._order_manager.get_order_status(
+            trading_pair=tracked_order.trading_pair, client_order_id=blxr_client_order_id
+        )
+
+        timestamp = time.time()
+        order_status = OrderStatus.OS_UNKNOWN
+        if len(order_status_info) != 0:
+            timestamp = order_status_info[-1].timestamp
```

Comment:
why `-1`, not and `0`?

Reply:
The endpoint gives all the trade updates for an order (as requested by `_all_trade_updates_for_order`, so the last one gives the most recent one

Reply:
ah, I see. :+1: